src/AdvanceSteelNodes/Util/Property.cs

```diff
@@ -23,6 +23,114 @@ internal Property()
 		{
 		}
 
+    /// <summary>
+    /// Get Single Advance Steel Property from Advance Steel Object
+    /// </summary>
+    /// <param name="steelObject"> Selected Advance Steel Object</param>
+    /// <param name="propertyType"> Advance Steel Property</param>
+    /// <returns></returns>
+    public static Object GetPropertyByElement(SteelDbObject steelObject, string propertyType)
+    {
+      Object ret = null;
+      using (var ctx = new SteelServices.DocContext())
+      {
+        ASProperty extractionProperty = Utils.GetProperty(propertyType, ePropertyDataOperator.Get);
+        if (extractionProperty != null)
+        {
+          if (extractionProperty.GetValueFromSteelDBObject(steelObject))
+          {
+            ret = extractionProperty.PropValue;
+          }
+        }
+        else
+        {
+          throw new System.Exception("InValid Property");
+        }
+      }
+      return ret;
+    }
+
+    /// <summary>
+    /// Generate a List of Advance Steel Object Properties based on Property Types
+    /// </summary>
+    /// <param name="steelObject"> Selected Advance Steel Object</param>
+    /// <param name="propertyTypes"> List of Property type per Object Type</param>
+    /// <param name="asStringForReference"> Displays the data as String</param>
+    /// <returns></returns>
+    public static Object GetPropertiesByElement(SteelDbObject steelObject, 
```

Comment:
why not return List<ASProperty> type? asStringForReference option should be a a method in ASProperty and removed from here.